web/src/compose_setup.js

```diff
@@ -594,7 +594,9 @@ export function initialize() {
 
     $("#compose_recipient_box").on("click", "#recipient_box_clear_topic_button", () => {
         const $input = $("input#stream_message_recipient_topic");
-        $input.val("");
+        compose_actions.start({trigger: "clear topic button", message_type: "stream"});
+        // We need to manually trigger focus so as to reliably
+        // tap into the focus-event logic below.
         $input.trigger("focus");
```

Comment:
I think this manual trigger should be able to be replaced by having `get_focus_area` return the right input with this trigger value?

Reply:
Also, does switching to `compose_actions.start` have other side effects, either involving restoring drafts or content already present in the compose box when you hit this button?

Reply:
Hmmmm. So, the left sidebar will clear the entire compose box when it is clicked. On `main`, the New Topic button in the recipient row clears only the topic box itself, leaving any message--even one restored from a draft--in place.

So rather than get all fancy with calling `compose.start()`, I've opted in the commit just pushed to simply make sure we're calling `compose_recipient.update_narrow_to_recipient_visibility()` when the clear-topic button is clicked.

Reply:
> I think this manual trigger should be able to be replaced by having `get_focus_area` return the right input with this trigger value?

Because I've returned to the simple box-clearing logic, I don't think we need to worry about this now.